packages/perseus/src/widgets/interactive-graphs/utils.ts

```diff
@@ -70,3 +70,27 @@ export function isUnlimitedGraphState(
         (state.type === "polygon" && state.numSides === "unlimited")
     );
 }
+
+/**
+ * Replace all text outside of the $ TeX blocks with `\\text{...}`
+ * This way, the entire resulting string can be rendered within <TeX>
+ * and the text outside of the $ blocks will be non-TeX text.
+ */
+export function replaceOutsideTeX(mathString) {
+    let currentlyTeX = mathString[0] === "$";
+    let result = "";
+
+    const splitString = mathString.split("$").filter((part) => part !== "");
+
+    for (let i = 0; i < splitString.length; i++) {
+        const part = splitString[i];
+        if (currentlyTeX) {
+            result += part;
+        } else {
+            result += `\\text{${part}}`;
+        }
+        currentlyTeX = !currentlyTeX;
+    }
+
+    return result;
+}
```

Comment:
I tried doing some regex stuff along with `mathString.replace()` but it got convoluted and kept letting edge cases slip through the cracks. I decided to go back to basics and stick to what works, and this was the resulting code.